include/pybind11/pytypes.h

```diff
@@ -1568,7 +1580,7 @@ template <typename D>
 str_attr_accessor object_api<D>::doc() const { return attr("__doc__"); }
 
 template <typename D>
-handle object_api<D>::get_type() const { return (PyObject *) Py_TYPE(derived().ptr()); }
+handle object_api<D>::get_type() const {return type::handle_of(*this);}
```

Comment:
```suggestion
handle object_api<D>::get_type() const { return type::handle_of(*this); }
```

Reply:
And add a deprecation, I thought, you were planning?

Reply:
I pulled back because I thought our policy was to not deprecate as soon as a replacement is available; but since this was never documented, it's probably safe to do so.